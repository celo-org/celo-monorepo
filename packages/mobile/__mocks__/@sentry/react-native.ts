const Sentry = {
  config: () => ({ install: jest.fn() }),
  setTagsContext: jest.fn(),
  captureException: jest.fn(),
}

module.exports = Sentry
