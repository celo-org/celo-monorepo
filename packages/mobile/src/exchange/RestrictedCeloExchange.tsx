import Button, { BtnSizes, BtnTypes } from '@celo/react-components/components/Button'
import ItemSeparator from '@celo/react-components/components/ItemSeparator'
import fontStyles from '@celo/react-components/styles/fonts'
import { Spacing } from '@celo/react-components/styles/styles'
import React from 'react'
import { useTranslation } from 'react-i18next'
import { StyleSheet, Text, View } from 'react-native'
import { Namespaces } from 'src/i18n'
import InfoIcon from 'src/icons/InfoIcon'

interface Props {
  onPressWithdraw: () => void
}

export default function RestrictedCeloExchange({ onPressWithdraw }: Props) {
  const { t } = useTranslation(Namespaces.exchangeFlow9)

  return (
    <>
      <View style={styles.topSpacing} />
      <ItemSeparator />
      <View style={styles.textContainer}>
        <Text style={styles.text}>{t('countryRestrictedExchange')}</Text>
        <View style={styles.infoIconContainer}>
          <InfoIcon size={14} />
        </View>
      </View>
      <ItemSeparator />
      <Button
        size={BtnSizes.FULL}
        text={t('withdrawCelo')}
        onPress={onPressWithdraw}
        style={styles.button}
        type={BtnTypes.TERTIARY}
        testID="WithdrawCELO"
      />
    </>
  )
}

const styles = StyleSheet.create({
  topSpacing: {
    marginTop: Spacing.Thick24,
  },
  textContainer: {
    marginVertical: Spacing.Regular16,
    marginHorizontal: Spacing.Regular16,
    flexDirection: 'row',
    alignItems: 'center',
    justifyContent: 'center',
  },
  text: {
    flexShrink: 1,
    ...fontStyles.small,
  },
  infoIconContainer: {
    marginLeft: Spacing.Smallest8,
  },
  button: {
    marginVertical: Spacing.Regular16,
    marginHorizontal: Spacing.Regular16,
  },
})
