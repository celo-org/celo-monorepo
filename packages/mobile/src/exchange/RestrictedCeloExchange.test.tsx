import * as React from 'react'
import 'react-native'
import { fireEvent, render } from 'react-native-testing-library'
import RestrictedCeloExchange from 'src/exchange/RestrictedCeloExchange'

describe('RestrictedCeloExchange', () => {
  it('renders correctly', () => {
    const onPressWithdraw = jest.fn()
    const tree = render(<RestrictedCeloExchange onPressWithdraw={onPressWithdraw} />)
    expect(tree).toMatchSnapshot()
    expect(onPressWithdraw).not.toHaveBeenCalled()

    fireEvent.press(tree.getByTestId('WithdrawCELO'))
    expect(onPressWithdraw).toHaveBeenCalled()
  })
})
