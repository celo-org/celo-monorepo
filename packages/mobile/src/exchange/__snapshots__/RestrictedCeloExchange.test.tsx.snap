// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RestrictedCeloExchange renders correctly 1`] = `
Array [
  <View
    style={
      Object {
        "marginTop": 24,
      }
    }
  />,
  <View
    style={
      Object {
        "backgroundColor": "#EDEEEF",
        "height": 1,
        "marginHorizontal": 16,
      }
    }
  />,
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
        "marginHorizontal": 16,
        "marginVertical": 16,
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#2E3338",
          "flexShrink": 1,
          "fontFamily": "Inter-Regular",
          "fontSize": 14,
          "lineHeight": 18,
        }
      }
    >
      countryRestrictedExchange
    </Text>
    <View
      style={
        Object {
          "marginLeft": 8,
        }
      }
    >
      <svg
        fill="none"
        height={14}
        style={Object {}}
        viewBox="0 0 16 16"
        width={14}
        xmlns="http://www.w3.org/2000/svg"
      >
        <circle
          cx="8"
          cy="8"
          r="7"
          stroke="#2E3338"
          stroke-width="1.25"
          style={Object {}}
        />
        <path
          d="M8 12V7M8 6V5"
          stroke="#2E3338"
          stroke-width="1.25"
          style={Object {}}
        />
      </svg>
    </View>
  </View>,
  <View
    style={
      Object {
        "backgroundColor": "#EDEEEF",
        "height": 1,
        "marginHorizontal": 16,
      }
    }
  />,
  <View
    style={
      Array [
        Object {
          "flexDirection": "column",
        },
        Object {
          "marginHorizontal": 16,
          "marginVertical": 16,
        },
      ]
    }
  >
    <View
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          Object {
            "borderRadius": 100,
          },
        ]
      }
    >
      <View
        accessible={true}
        focusable={true}
        nativeBackgroundAndroid={
          Object {
            "attribute": "selectableItemBackground",
            "type": "ThemeAttrAndroid",
          }
        }
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#EEB93C",
            "flexGrow": 1,
            "height": 48,
            "justifyContent": "center",
            "opacity": undefined,
            "paddingHorizontal": 24,
            "paddingVertical": 5,
          }
        }
        testID="WithdrawCELO"
      >
        <Text
          style={
            Object {
              "color": "#FFFFFF",
              "fontFamily": "Inter-SemiBold",
              "fontSize": 16,
              "lineHeight": 22,
            }
          }
        >
          withdrawCelo
        </Text>
      </View>
    </View>
  </View>,
]
`;
